### DISCUSSION TITLE
Synology - postdownload script to index the downloaded file?

### DISCUSSION THREAD

**Discussion by dorianps** [haugene/docker-transmission-openvpn#2644] [Q&A]
This is a great docker solution. However, the downloaded videos are not automatically indexed in synology's Video Station. This used to be a constant problem with the host transmission some years ago, but was fixed in the Transmission release of Synology. Now the problem comes back.

Does anyone have a ready solution to force all mp4/mkv files to be indexed by Video Station?

My idea is to use a transmission post download script, but it is getting a bit complicated to call synoindex of the host system from the docker container. I found another docker solution that may serve as intermediary and calls the host synoindex (https://github.com/racklin/simple-synoindex-server), but this sounds again quite complicated.

I think many other people have encountered the problem and probably someone has a solution. I appreciate if there is something ready to go.

**Comment by ilike2burnthing** [comment]
Have you configured Video Station to index the completed downloads folder?

**Comment by dorianps** [comment]
Absolutely yes. If I even move the file through samba (Windows mapped
folder) Video Station picks it up and indexes it. But nothing happens when
transmission within this docker just finishes downloading files...

**Reply by ilike2burnthing** [reply]
Well I found an unanswered post from 2019 about the exact issue:
https://community.synology.com/enu/forum/1/post/124019

A probably related post also from 2019:
https://community.synology.com/enu/forum/1/post/130719

And a possibly related post from 2009:
https://community.synology.com/enu/forum/17/post/13831

What do you have configured for `TRANSMISSION_DOWNLOAD_DIR` and `TRANSMISSION_INCOMPLETE_DIR` and what volumes do you have mapped?

**Comment by dorianps** [comment]
Sorry for the delay.

I don't think the output folder matters as long as Video Station is indexing it. And for sure it does if not for newly completed torrents. This is the full command for running the container with all the info you may need:

```
VPNUSER=XXXXX
VPNPASS=XXXXX
OUTPUT_FOLDER=/volume2/MeaCulpa/Download/
CONFIG_FOLDER=/volume2/MeaCulpa/Download/DONOTDELETE_config/


docker run -d \
              --restart unless-stopped \
              --cap-add=NET_ADMIN  \
              -v $OUTPUT_FOLDER:/data \
              -v $CONFIG_FOLDER:/config \
              -e OPENVPN_PROVIDER=FASTESTVPN \
              -e OPENVPN_CONFIG=France-UDP,Belgium1-UDP,Brazil1-UDP,Bulgaria-UDP  \
              -e OPENVPN_USERNAME=$VPNUSER \
              -e OPENVPN_PASSWORD=$VPNPASS \
              -e LOCAL_NETWORK=192.168.5.0/24 \
              -e TRANSMISSION_DOWNLOAD_DIR=/data \
              -e TRANSMISSION_INCOMPLETE_DIR=/data/incomplete \
              -e TRANSMISSION_RATIO_LIMIT_ENABLED=true \
              -e TRANSMISSION_RATIO_LIMIT=0 \
              -e TRANSMISSION_SPEED_LIMIT_UP_ENABLED=true \
              -e TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=true \
              -e TRANSMISSION_RPC_USERNAME=XXXXXXXXXX \
              -e TRANSMISSION_RPC_PASSWORD=XXXXXXXXXXX \
              -e TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false \
              -e TRANSMISSION_SCRIPT_TORRENT_DONE_ENABLED=true \
              -e TRANSMISSION_SCRIPT_TORRENT_DONE_FILENAME=/config/transmission_post_download_script.sh \
              -e PUID=1026 \
              -e PGID=100 \
              --dns 8.8.8.8 \
              --dns 8.8.4.4 \
              --log-driver json-file \
              --log-opt max-size=10m \
              -p 9091:9091 \
              --name transmission \
              haugene/transmission-openvpn:master
#-----------
``

**Comment by dorianps** [comment]
There is a solution looks like, using VS-transmission docker container.

https://github.com/heimdall-syno/VS-Transmission

The example uses this very transmission-openvpn. VS-transmission needs some learning how to set up though, and hopefully works on DSM 7+. I was hoping someone here has experience with the problem and the solution.

**Reply by ilike2burnthing** [reply]
I saw that, but the description seems like it wouldn't help if you're using volume mapping, which seems like an infinitely simpler solution - `VS-SynoIndex is a server-based wrapper for Synology's SynoIndex to enable indexing video files even from winthin [sic] docker container.`

So, to be clear, `/volume2/MeaCulpa/Download/` is indexed by Video Station? If so, then the cause of your problem is probably related to the issue in the links I posted above.

**Comment by dorianps** [comment]
Isn't volume mapping above in my command the same folder mapping you refer
to? I.e., volume binding.

**Reply by ilike2burnthing** [reply]
It is, that's where I got `/volume2/MeaCulpa/Download/` from.

**Reply by dorianps** [reply]
Yes, `/volume2/MeaCulpa/Download/` is indexed by Video Station. If I am not mistaken, the links you posted conclude that the problem is that any file created internally or via ssh is not picked up for indexing by synoindex. If files are moved or created via samba, then synoindex is called and they are added to the library.

So, in short, do you see any solution except calling synoindex, or perhaps a post download script that calls synoindex? Any ready solution to solve the problem created by transmission?

**Reply by ilike2burnthing** [reply]
Post-download script would seem to be the solution, yes.
